src/logplex_logs_rest.erl

```diff
@@ -259,3 +288,14 @@ respond(Code, Text, Req, State) ->
     Req3 = cowboy_req:set_resp_body(Text, Req2),
     {ok, Req4} = cowboy_req:reply(<<(integer_to_binary(Code))/binary, " ", Text/binary>>, Req3),
     {halt, Req4, State}.
+
+decompress_body(CompressedBody, ZPort) ->
+    try
+        [Inflated] = zlib:inflate(ZPort, CompressedBody),
+        %% reset rather than end so we can call this function again if
+        %% need be.
+        ok = zlib:inflateReset(ZPort),
+        {ok, Inflated}
+    catch _Class:Reason ->
+            {error, Reason}
```

Comment:
Is there any risk the zlib port is in a bad state in case of failure? Does it need to be restarted at a higher level then?

The other question I have more generally is whether that zlib port needs to be started for all reqs or only those that have the compression body. If the port is not kept alive cross-request, maybe we're looking at far more open/close cycles than necessary if say, 10% of reqs only do compression?